src/main/scala/is/hail/annotations/aggregators/RegionValueCounterAggregator.scala

```diff
@@ -0,0 +1,282 @@
+package is.hail.annotations.aggregators
+
+import is.hail.annotations.{Annotation, Region, RegionValueBuilder, SafeRow}
+import is.hail.expr.types.Type
+import is.hail.utils._
+
+import scala.collection.mutable
+
+class RegionValueCounterBooleanAggregator extends RegionValueAggregator {
+  private val a = Array(0L, 0L, 0L)
+
+  def seqOp(region: Region, x: Boolean, missing: Boolean) {
+    if (missing)
+      a(2) += 1
+    else if (x)
+      a(1) += 1
+    else
+      a(0) += 1
+  }
+
+  override def combOp(agg2: RegionValueAggregator) {
+    val other = agg2.asInstanceOf[RegionValueCounterBooleanAggregator]
+    a(0) += other.a(0)
+    a(1) += other.a(1)
+    a(2) += other.a(2)
+  }
+
+  override def result(rvb: RegionValueBuilder) {
+    rvb.startArray(a.count(_ != 0))
+
+    if (a(0) > 0) {
+      rvb.startStruct()
+      rvb.addBoolean(false)
+      rvb.addLong(a(0))
+      rvb.endStruct()
+    }
+
+    if (a(1) > 0) {
+      rvb.startStruct()
+      rvb.addBoolean(true)
+      rvb.addLong(a(1))
+      rvb.endStruct()
+    }
+
+    if (a(2) > 0) {
+      rvb.startStruct()
+      rvb.setMissing()
+      rvb.addLong(a(2))
+      rvb.endStruct()
+    }
+
+    rvb.endArray()
+  }
+
+  override def copy(): RegionValueCounterBooleanAggregator = new RegionValueCounterBooleanAggregator()
+
+  override def clear() {
+    a(0) = 0L
+    a(1) = 0L
+    a(2) = 0L
+  }
+}
+
+class RegionValueCounterIntAggregator extends RegionValueAggregator {
+  private val m = mutable.Map[Int, Long]()
```

Comment:
Suggestion: Since Map boxes, specializing all these versions doesn't gain us very much.  You'll still need versions with seqOps with the right types, but you might be able to have a single common base class with a Map[Any, Long].